index.js

```diff
@@ -177,4 +183,48 @@ async function installBundler(platform, rubyPrefix, engine, rubyVersion) {
   }
 }
 
+async function bundleInstall(platform, engine, version) {
+  // config
+  const path = 'vendor/bundle'
+  const hasGemfileLock = fs.existsSync('Gemfile.lock');
+  if (hasGemfileLock) {
+    await exec.exec('bundle', ['config', 'deployment', 'true'])
+  }
+  await exec.exec('bundle', ['config', 'path', path])
+
+  // cache key
+  const paths = [path]
+  const baseKey = `setup-ruby-bundle-install-${platform}-${engine}-${version}`
+  let key = baseKey
+  let restoreKeys
+  if (hasGemfileLock) {
+    key += `-${await common.hashFile('Gemfile.lock')}`
+    // If only Gemfile.lock we can reuse some of the cache (but it will keep old gem versions in the cache)
+    restoreKeys = [`${baseKey}-`]
+  } else {
+    // Only exact key, to never mix native gems of different platforms or Ruby versions
+    restoreKeys = []
+  }
+  console.log(`Cache key: ${key}`)
+
+  // restore cache & install
+  const cachedKey = await cache.restoreCache(paths, key, restoreKeys)
+  if (cachedKey) {
+    console.log(`Found cache for key: ${cachedKey}`)
+  }
+
+  let alreadyInstalled = false
+  if (cachedKey === key) {
+    const exitCode = await exec.exec('bundle', ['check'])
+    alreadyInstalled = (exitCode === 0)
+  }
+
+  if (!alreadyInstalled) {
+    await exec.exec('bundle', ['install', '--jobs', '4'])
```

Comment:
https://github.com/actions/cache/blob/9ab95382c899bf0953a0c6c1374373fc40456ffe/examples.md#ruby---bundler also have the `--retry` flag in its example (`--retry 3`)

I guess it doesn't hurt for this action to use it too?